File: java/demos/ps/maps/Tonoe.java
package demos.ps.maps;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static demos.ps.oo.PSGame.*;
import demos.ps.oo.City;
import demos.ps.oo.Dungeon;
import demos.ps.oo.Enemy;
import demos.ps.oo.PSGame;
import demos.ps.oo.Item;
import demos.ps.oo.PSGame.Flags;
import demos.ps.oo.PSLibEnemy.PS1Enemy;
import demos.ps.oo.PSLibItem.OriginalItem;
import demos.ps.oo.PSLibSound.PS1Sound;
import demos.ps.oo.PSMenu;
import demos.ps.oo.PSMenu.EntityClothes;
import demos.ps.oo.PSMenu.EntityType;
import demos.ps.oo.PSMenu.LargeEntity;
import demos.ps.oo.PSMenu.MotaCape;
import demos.ps.oo.PSMenu.Outcome;
import demos.ps.oo.PSMenu.Scene;
import demos.ps.oo.PSMenu.SpecialEntity;

public class Tonoe {

	private static final Logger log = LogManager.getLogger(Tonoe.class);

	public static void church() {
		PSMenu.startScene(Scene.CHURCH_VILLAGE, EntityType.MOTA_CAP, MotaCape.YELLOW);
		PSGame.Church(1);
		PSMenu.endScene();
	}	
	
	public static void food_shop() {
		PSMenu.startScene(Scene.SHOP_FOOD_VILLAGE, EntityType.MOTA_CAP, MotaCape.YELLOW);
		Shop(getString("Shop_Pharmacy_Welcome"), false, new Item[]{	getItem(OriginalItem.Inventory_Monomate),
																	getItem(OriginalItem.Inventory_Dimate),
																	getItem(OriginalItem.Inventory_Trimate)});
		PSMenu.endScene();
	}	
	public static void hand_shop() {
		PSMenu.startScene(Scene.SHOP_HAND_VILLAGE, EntityType.MOTA_NOCAP, MotaCape.GREEN);
		Shop(getString("Shop_Tool_Welcome"), true, new Item[]{	getItem(OriginalItem.Quest_Alsuline)});
		PSMenu.endScene();
	}	
	
	
	public static void chief_house() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.MOTA_CUSTOM, MotaCape.RED);
		PSMenu.Stext(getString("Tonoe_House_Chief"));
		PSMenu.endScene();		 
	}
	
	public static void house1() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.MOTA_NOCAP, MotaCape.YELLOW);
		if(PSMenu.Prompt(getString("Tonoe_House_1"), getYesNo()) == 1) {
			PSMenu.StextLast(getString("Tonoe_House_1Yes"));
		} else {
			PSMenu.StextLast(getString("Tonoe_House_1No"));
		}
		PSMenu.endScene();
	}
	public static void house2() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.MOTA_CAP, MotaCape.RED);
		PSMenu.Stext(getString("Tonoe_House_2"));
		PSMenu.endScene();
	}
	public static void house3() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.MOTA_MASK, MotaCape.GREEN);
		if(!PSGame.hasFlag(Flags.INFO_TONOE_DAUGHTER)) {
			PSMenu.Stext(getString("Tonoe_House_3"));	
		}
		else {
			PSMenu.Stext(getString("Tonoe_House_3_Rescue"));
			PSGame.playSound(PS1Sound.CURE);
			getParty().healAll(false);
		}
		PSMenu.endScene();
	}
	public static void house4() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.MOTA_MASK, MotaCape.BROWN);
		PSMenu.Stext(getString("Tonoe_House_4"));
		PSMenu.endScene();
	}
	public static void house5() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.MOTA_CUSTOM, MotaCape.GREEN);
		PSMenu.Stext(getString("Tonoe_House_5"));
		PSMenu.endScene();
	}
	
	public static void ent1() {
		EntStart();
		PSMenu.Stext(getString("Tonoe_People_Ent1"));
		EntFinish();
	}
	public static void ent2() {
		EntStart();
		PSMenu.Stext(getString("Tonoe_People_Ent2"));
		EntFinish();
	}
	public static void ent3() {
		EntStart();
		PSMenu.Stext(getString("Tonoe_People_Ent3"));
		EntFinish();
	}
	
	public static void exit() {
		mapswitch(Planet.MOTAVIA, City.TONOE.getX(), City.TONOE.getY());
	}

	public static void cave() {
		mapswitch(Dungeon.BLUEBERRY_MINE);
	}

	
	
}

File: src/demos/ps/maps/Tonoe.ts
/**
 * Tonoe Village Script
 * TypeScript port of Tonoe.java
 */

import { PSGame } from '../PSGame';
import { Flags } from '../game/GameData';
import { Planet, City, CityHelper } from '../game/City';
import { Dungeon } from '../game/Dungeon';
import { OriginalItem } from '../game/PSLibItem';
import { PS1Sound } from '../game/PSLibSound';
import { PSSceneType, EntityType, EntityClothes, MotaCape, PSMenu } from '../PSMenu';
import { PSMenuShop } from '../PSMenuShop';

// MotaCape values act as the "clothes" frame offset, same numeric slot as EntityClothes.
const cape = (c: MotaCape): EntityClothes => c as unknown as EntityClothes;

export class Tonoe {

  public static async church(): Promise<void> {
    await PSMenu.startScene(PSSceneType.CHURCH_VILLAGE, EntityType.MOTA_CAP, cape(MotaCape.YELLOW));
    await PSGame.Church(1);
    await PSMenu.endScene();
  }

  public static async food_shop(): Promise<void> {
    await PSMenu.startScene(PSSceneType.SHOP_FOOD_VILLAGE, EntityType.MOTA_CAP, cape(MotaCape.YELLOW));
    await PSMenuShop.Shop(PSGame.getString("Shop_Pharmacy_Welcome"), false, PSGame.getParty(), [
      PSGame.getItem(OriginalItem.Inventory_Monomate),
      PSGame.getItem(OriginalItem.Inventory_Dimate),
      PSGame.getItem(OriginalItem.Inventory_Trimate)
    ]);
    await PSMenu.endScene();
  }

  public static async hand_shop(): Promise<void> {
    await PSMenu.startScene(PSSceneType.SHOP_HAND_VILLAGE, EntityType.MOTA_NOCAP, cape(MotaCape.GREEN));
    await PSMenuShop.Shop(PSGame.getString("Shop_Tool_Welcome"), true, PSGame.getParty(), [
      PSGame.getItem(OriginalItem.Quest_Alsuline)
    ]);
    await PSMenu.endScene();
  }

  public static async chief_house(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.MOTA_CUSTOM, cape(MotaCape.RED));
    await PSMenu.Stext(PSGame.getString("Tonoe_House_Chief"));
    await PSMenu.endScene();
  }

  public static async house1(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.MOTA_NOCAP, cape(MotaCape.YELLOW));
    if (await PSMenu.Prompt(PSGame.getString("Tonoe_House_1"), PSGame.getYesNo()) === 1) {
      await PSMenu.StextLast(PSGame.getString("Tonoe_House_1Yes"));
    } else {
      await PSMenu.StextLast(PSGame.getString("Tonoe_House_1No"));
    }
    await PSMenu.endScene();
  }

  public static async house2(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.MOTA_CAP, cape(MotaCape.RED));
    await PSMenu.Stext(PSGame.getString("Tonoe_House_2"));
    await PSMenu.endScene();
  }

  public static async house3(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.MOTA_MASK, cape(MotaCape.GREEN));
    if (!PSGame.hasFlag(Flags.INFO_TONOE_DAUGHTER)) {
      await PSMenu.Stext(PSGame.getString("Tonoe_House_3"));
    } else {
      await PSMenu.Stext(PSGame.getString("Tonoe_House_3_Rescue"));
      PSGame.playSound(PS1Sound.CURE);
      PSGame.getParty().healAll(false);
    }
    await PSMenu.endScene();
  }

  public static async house4(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.MOTA_MASK, cape(MotaCape.BROWN));
    await PSMenu.Stext(PSGame.getString("Tonoe_House_4"));
    await PSMenu.endScene();
  }

  public static async house5(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.MOTA_CUSTOM, cape(MotaCape.GREEN));
    await PSMenu.Stext(PSGame.getString("Tonoe_House_5"));
    await PSMenu.endScene();
  }

  public static async ent1(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Tonoe_People_Ent1"));
    PSGame.EntFinish();
  }

  public static async ent2(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Tonoe_People_Ent2"));
    PSGame.EntFinish();
  }

  public static async ent3(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Tonoe_People_Ent3"));
    PSGame.EntFinish();
  }

  public static exit(): void {
    PSGame.mapswitchToPlanet(Planet.MOTAVIA, CityHelper.getX(City.TONOE), CityHelper.getY(City.TONOE));
  }

  public static async cave(): Promise<void> {
    await PSGame.mapswitchToDungeon(Dungeon.BLUEBERRY_MINE);
  }
}
